### PR TITLE
[ios] Enable/Disable sku token handling

### PR DESCRIPTION
**Pull request by julianrex**
Disables the default behavior of PR #14421 during the pre-release period to facilitate testing.

### CODE REVIEW

**Comment by friedbunny** [comment]
@julianrex like you mentioned in chat, I think a plist key would be a better way to handle this, since:
- We want to ensure that the telemetry SKU token parameter is set for the initial turnstile, which will be easier to guarantee if we’re not worried about arbitrary settings of `MGLAccountManager.accountsSDKEnabled`.
- We don’t need to support post-launch toggling of SKU token usage. (Right, @zugaldia?)

**Comment by zugaldia** [comment]
> We don’t need to support post-app-launch toggling of SKU token usage. (Right, @zugaldia?)

Correct - this is scoped to the pre-releases period only (effectively reverting this change before final release).

**Comment by julianrex** [comment]
Should make the code simpler too.

File: platform/darwin/src/MGLAccountManager.m [hunk 1]
```diff
@@ -41,7 +42,21 @@ + (void)load {
     }
 
 #if TARGET_OS_IPHONE || TARGET_OS_SIMULATOR
-    self.skuToken = [MBXSKUToken skuToken];
+    if (accessToken.length) {
```
Comment (friedbunny) [inline platform/darwin/src/MGLAccountManager.m]: This duplicates some lines just above here.

File: platform/darwin/src/http_file_source.mm [hunk 2]
```diff
@@ -208,10 +208,19 @@ void cancel() {
...
+            
+            NSMutableArray *newQueryItems = [NSMutableArray arrayWithArray:@[
+                [NSURLQueryItem queryItemWithName:@"events" value:@"true"]
+                ]];
+
+            // Only add the token if we have enabled the accounts SDK
+            if (MGLAccountManager.isAccountsSDKEnabled)
+            {
+                NSString *skuToken = MGLAccountManager.skuToken;
+                if (skuToken) {
```
Comment (friedbunny) [inline platform/darwin/src/http_file_source.mm]: Checking for `nil` here feels redundant to me, given that `MGLAccountManager.skuToken` shouldn't return `nil` if `MGLAccountManager.isAccountsSDKEnabled = YES`.

**Pull request merged**